src/dec128.rs

```diff
@@ -104,16 +104,15 @@ struct d128Visitor;
 impl serde::de::Visitor for d128Visitor {
     type Value = d128;
 
-    fn visit_str<E>(&mut self, s: &str) -> Result<d128, E>
+    fn expecting(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "a d128 value")
+    }
+
+    fn visit_str<E>(self, s: &str) -> Result<d128, E>
         where E: serde::de::Error
     {
-        let d = match d128::from_str(s) {
-            Ok(d) => { d }
-            Err(_) => { return Err(E::custom("Failed to convert to d128")); }
-        };
-
-
-        Ok(d)
+        use serde::de::Unexpected;
+        d128::from_str(s).map_err(|_| E::invalid_type(Unexpected::Str(s), &"a d128 value"))
```

Comment:
I would prefer to use `E::invalid_value(Unexpected::Str(s), &self)`:

- Avoid repeating the message you already have in `fn expecting`.
- The type (string) is correct but the value contained in that string is the problem.